src/pyobo/sources/gtdb.py

```diff
@@ -0,0 +1,111 @@
+    "c__": Reference(prefix="TAXRANK", identifier="0000002", name="class"),
+    "o__": Reference(prefix="TAXRANK", identifier="0000003", name="order"),
+    "f__": Reference(prefix="TAXRANK", identifier="0000004", name="family"),
+    "g__": Reference(prefix="TAXRANK", identifier="0000005", name="genus"),
+    "s__": Reference(prefix="TAXRANK", identifier="0000006", name="species"),
+}
+
+GTDB_AR_URL = "https://data.gtdb.ecogenomic.org/releases/latest/ar53_metadata.tsv.gz"
+GTDB_BAC_URL = "https://data.gtdb.ecogenomic.org/releases/latest/bac120_metadata.tsv.gz"
+
+logger = logging.getLogger(__name__)
+
+
+class GTDBGetter(Obo):
+    """An ontology representation of the GTDB taxonomy."""
+
+    ontology = bioversions_key = PREFIX
+    typedefs = [has_taxonomy_rank]
+
+    def iter_terms(self, force: bool = False) -> Iterable[Term]:
+        """Iterate over terms in the ontology."""
+        return iter_terms(version=self._version_or_raise, force=force)
+
+
+def _parse_gtdb_taxonomy(tax_string: str) -> list[tuple[str, str]]:
+    """Parse GTDB taxonomy string into (level, name) tuples."""
+    taxa = []
+    parts = [p.strip() for p in tax_string.split(";") if p.strip()]
+
+    for part in parts:
+        if len(part) < 4 or "__" not in part:
+            logger.warning(f"Malformed taxon string: {part}")
+            continue
+        level = part[:3]
+        name = part[3:].strip()
+        if level in LEVEL_ORDER and name:
+            taxa.append((level, name))
+        else:
+            logger.warning(f"Invalid taxonomic level or name in part: {part}")
+    return taxa
+
+
+def iter_terms(version: str, force: bool = False) -> Iterable[Term]:
+    """Iterate over GTDB terms."""
+    ar_path = ensure_path(PREFIX, url=GTDB_AR_URL, version=version, force=force)
+    bac_path = ensure_path(PREFIX, url=GTDB_BAC_URL, version=version, force=force)
+    columns = ["gtdb_taxonomy", "ncbi_taxid"]
+    for path_name, path in [
+        ("ar", ar_path),
+        ("bac", bac_path),
+    ]:
+        df = pd.read_csv(path, sep="\t", dtype=str)
+        for tax_string, ncbitaxon_id in tqdm(
+            df[columns].values, desc=f"[{PREFIX}] processing {path_name}", unit_scale=True
+        ):
+            yield from _process_row(tax_string, ncbitaxon_id)
+
+
+def _process_row(tax_string, ncbitaxon_id) -> Iterable[Term]:
+    if not isinstance(tax_string, str):
+        logger.warning(f"Invalid taxonomy string: {tax_string}")
+        return None
+
+    taxa = _parse_gtdb_taxonomy(tax_string)
+    if not taxa:
+        logger.warning(f"No valid taxa found in: {tax_string}")
+        return None
+
+    parent_reference = None
+    for level, name in taxa:
+        identifier = f"{level}{name.replace(' ', '_')}"
+        term = Term(
+            reference=Reference(prefix=PREFIX, identifier=identifier, name=name),
+        )
+        if taxrank_reference := LEVEL_TO_TAXRANK.get(level):
+            term.append_property(has_taxonomy_rank, taxrank_reference)
```

Comment:
Note, I took out the description - this was better served by using a structured annotation

Reply:
As I mentioned above, there are over 50 columns in that source file, with my first attempt at creating a PyOBO module, the intent was to come up with an MVP with the minimal goal of establishing the terms for gtdb taxon names and dbxref to ncbi tax ids, to fulfill the needs of our project that led me to explore pyobo and decide to use it to create the ontologies we need that are currently unavailable. I am happy to engage in a discussion about what other fields we could consider including, something like biosample project and sample columns i ssomething i actually had not thought about not sure i have seen that in other ontologies, but I am new to exploring this type of data.